go/vt/mysqlctl/replication_test.go

```diff
@@ -80,3 +88,586 @@ func TestRedactPassword(t *testing.T) {
   PASSWORD = '****'
 `)
 }
+
+func TestWaitForReplicationStart(t *testing.T) {
+	db := fakesqldb.New(t)
+	fakemysqld := NewFakeMysqlDaemon(db)
+
+	defer func() {
+		db.Close()
+		fakemysqld.Close()
+	}()
+
+	err := WaitForReplicationStart(fakemysqld, 2)
+	assert.NoError(t, err)
+}
```

Comment:
This is not a great test. We should be testing for the case where we get errors too. Otherwise we could end up in the same situation wherein the function never returns an error. The tests would pass even then and won't catch the bug.

Reply:
done.